### PR TITLE
fix: ignoreScripts shoud be true when lockfileonly set true in the install process

### PR DESCRIPTION
**Pull request by fireairforce**
closes: https://github.com/pnpm/pnpm/issues/3750

### CODE REVIEW

File: packages/plugin-commands-import/src/import.ts [hunk 1]
```diff
@@ -34,8 +34,9 @@ export function help () {
 export const commandNames = ['import']
 
 export async function handler (
-  opts: CreateStoreControllerOptions & Omit<InstallOptions, 'storeController' | 'lockfileOnly' | 'preferredVersions'>
+  opts: CreateStoreControllerOptions & Omit<InstallOptions, 'storeController' | 'lockfileOnly' | 'preferredVersions' | 'ignoreScripts'>
```
Comment (zkochan) [inline packages/plugin-commands-import/src/import.ts]: why was this added if it is not used?

File: packages/plugin-commands-import/src/import.ts [hunk 2]
```diff
@@ -34,8 +34,9 @@ export function help () {
...
 ) {
+  console.log('test case')
```
Comment (zkochan) [inline packages/plugin-commands-import/src/import.ts]: this is not needed

File: packages/plugin-commands-import/test/index.ts [hunk 3]
```diff
@@ -41,6 +41,7 @@ const DEFAULT_OPTS = {
   userAgent: 'pnpm',
   useRunningStoreServer: false,
   useStoreServer: false,
+  ignoreScripts: true,
```
Comment (zkochan) [inline packages/plugin-commands-import/test/index.ts]: I don't understand what this changes if the value of ignoreScripts is hard coded in src/import.ts

File: packages/plugin-commands-import/src/import.ts (hunk 2 above)
Comment (fireairforce) [reply packages/plugin-commands-import/src/import.ts]: oh, sorry, i forget delete it..

File: packages/plugin-commands-import/test/index.ts (hunk 3 above)
Comment (fireairforce) [reply packages/plugin-commands-import/test/index.ts]: Because  `pnpm import` only generate pnpm-lock.yaml but not install the `node_modules` directory, so i think it maybe  not need to execute the scripts... pls see https://github.com/pnpm/pnpm/issues/3750

File: packages/plugin-commands-import/src/import.ts (hunk 1 above)
Comment (fireairforce) [reply packages/plugin-commands-import/src/import.ts]: the value is hard code here...

**Comment by zkochan** [comment]
Actually, this needs to be changed in supi. If `lockfileOnly` is `true`, ignoreScripts should be set to true as well. If ignoreScripts is set to true, an exception should be thrown. So `pnpm install --no-ignore-scripts --lockfile-only` should actually throw an exception

**Comment by fireairforce** [comment]
ok, i will make some changes in supi.

**Comment by fireairforce** [comment]
I update the supi, @zkochan , PTAL if you are free

File: packages/supi/src/install/extendInstallOptions.ts [hunk 4]
```diff
@@ -174,6 +174,14 @@ export default async (
...
     extendedOpts.userAgent = `${extendedOpts.packageManager.name}/${extendedOpts.packageManager.version} ${extendedOpts.userAgent}`
   }
+  if (extendedOpts.lockfileOnly && !extendedOpts.ignoreScripts) {
+    if (opts.ignoreScripts === false) {
+      throw new PnpmError('CONFIG_CONFLICT_LOCKFILE_ONLY_WITH_NO_IGNORE_SCRIPTS',
+        'Cannot execute the scripts because lockfileOnly is set to false')
+    } else {
+      extendedOpts.ignoreScripts = true
+    }
+  }
```
Comment (zkochan) [inline packages/supi/src/install/extendInstallOptions.ts]: Seems like it will have to be like this, otherwise the tests will fail.

```suggestion
  if (extendedOpts.lockfileOnly) {
    extendedOpts.ignoreScripts = true
  }
```
